spec/Schema.spec.js

```diff
@@ -1367,6 +1367,24 @@ describe('SchemaController', () => {
   });
 });
 
+describe('Class Level Permissions', () => {
+  it('does not require addField for nested modification (#7371)', async () => {
+    const testSchema = new Parse.Schema('test_7371');
+    testSchema.setCLP({
+      create: { ['*']: true },
+      update: { ['*']: true },
+      addField: {},
+    });
+    testSchema.addObject('a');
+    await testSchema.save();
+    const obj = new Parse.Object('test_7371');
+    obj.set('a', { b: 1 });
+    await obj.save();
+    obj.set('a.b', 2);
```

Comment:
Could you split this into two distinct tests please, one where a new key is added, one where an existing key is modified, and change the test descriptions accordingly? That indicates to future contributors that there are two scenarios being tested.

Reply:
Neither of the two calls to `set` add a new field to the class.
I don't think adding a new key to a nested object should be subject to the addField permission, because otherwise it would be impossible to create any non-empty nested object in a class without having the addField permission.

Reply:
Maybe I'm still misunderstanding the issue.

Is the issue that changing the value of a key of a Parse Object field of type `Object`?

For example of field `a` changing the value of key `b`:
```
Parse.Object = {
  a: {
       b: 0
     }
}
```

Reply:
That is correct

Reply:
Does the issue also occur if the key is not set? In other words, if you do:

```
obj.set('a', {});
```

instead of:

```
obj.set('a', { 'b': 1 });
```

Reply:
I am able to manually verify that the issue also occurs in this case.
I see your point now, but the current test does not cover this scenario, and to the best of my knowledge, these scenarios do not correspond to different code paths.
Should I add it as a separate test?

Reply:
I think it could be beneficial to add that as a separate test, because we cannot foresee what modifications contributors will make in the future. The additional test will ensure that the permissions won't be required for that case.

Reply:
Done. I have also moved the new tests to a different file that felt like a more suitable place.

Reply:
Would you have further feedback @mtrezza ?

Reply:
That was good, thanks!